ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.cpp

```diff
@@ -0,0 +1,235 @@
+#include "MockVnodeAndMount.hpp"
+#include "../PrjFSKext/VnodeUtilities.hpp"
+#include "KextMockUtilities.hpp"
+#include <unordered_map>
+#include <sys/errno.h>
+
+using std::string;
+
+typedef std::unordered_map<string, VnodeWeakPointer> PathToVnodeMap;
+typedef std::unordered_map<vnode_t, VnodeWeakPointer> WeakVnodeMap;
+
+static PathToVnodeMap s_vnodesByPath;
+static WeakVnodeMap s_allVnodes;
+
+MountPointer mount::Create(const char* fileSystemTypeName, fsid_t fsid, uint64_t initialInode)
+{
+    MountPointer result(new mount{});
+    assert(strlen(fileSystemTypeName) + 1 < sizeof(result->statfs.f_fstypename));
+    result->statfs.f_fsid = fsid;
+    result->nextInode = initialInode;
+    strlcpy(result->statfs.f_fstypename, fileSystemTypeName, sizeof(result->statfs.f_fstypename));
+    return result;
+}
+
+
+vnode::vnode(const MountPointer& mount) :
+    mountPoint(mount),
+    name(nullptr),
+    inode(mount->nextInode++)
+{
+}
+
+vnode::~vnode()
+{
+    assert(this->ioCount == 0);
+}
+
+VnodePointer vnode::Create(const MountPointer& mount, const char* path, vtype vnodeType)
+{
+    VnodePointer result(new vnode(mount));
```

Comment:
Minor: can we use `make_shared` here?

Reply:
The constructor for `vnode` is private, so [not without a workaround](https://stackoverflow.com/questions/8147027/how-do-i-call-stdmake-shared-on-a-class-with-only-protected-or-private-const). We can of course make the constructor public, but the reason it's private is because the `s_allVnodes` check won't work if someone accidentally creates an instance without going via `::Create()`.

Reply:
Missed that detail, thanks!